include/dlaf/util_math.h

```diff
@@ -18,8 +18,47 @@
 
 /// @file
 
-namespace dlaf {
-namespace util {
+namespace dlaf::util {
+
+// Interleaves two intervals of length `l` split in blocks of size `b` starting at offsets `o1` and
+// `o2` respectively and returns an array of indices where the splits have occured.
+//
+// o1
+//  │
+//  └► │   │   │   │   │
+//    ─┴───┴───┴───┴───┴───  ◄─┐
+//               ▲             │
+// o2      b ─┬──┘              l
+// │          │                │
+// └─►  │   │ ▼ │   │   │      │
+//    ──┴───┴───┴───┴───┴──  ◄─┘
+//
+inline std::vector<SizeType> interleaveSplits(SizeType l, SizeType b, SizeType o1, SizeType o2) {
+  DLAF_ASSERT(l > 0, l);
+  DLAF_ASSERT(b > 0, b);
+  DLAF_ASSERT(o1 >= 0, o1);
+  DLAF_ASSERT(o2 >= 0, o2);
```

Comment:
I think (according to tests and implementation) that offset should be smaller than b.
Please complete the assertions or add corresponding test cases.